newdle/api.py

```diff
@@ -77,6 +81,38 @@ def require_token():
     g.user = user
 
 
+@api.route('/avatar/<email>')
+@allow_anonymous
+def user_avatar(email):
+    size = request.args.get('size')
+    email_hex = hashlib.md5(email.encode()).hexdigest()
+    # make gravat return 404 HTTP code instead of a default image
+    query_args = {'d': '404'}
+    if size is not None:
+        query_args['s'] = size
+    gravatar_url = (
+        f'https://gravatar.com/avatar/{quote(email_hex)}?' f'{urlencode(query_args)}'
+    )
+    resp = requests.get(gravatar_url)
```

Comment:
what about forwarding headers from the client in order to handle client-side caching (if-modified-since) nicely?

Reply:
yeah, would you forward all of them?

Reply:
forwarding headers relevant for caching should probably be enough... (i'd have a look at the headers your browser sends when you just open (no f5 reload, since both that and a hard ctrl+f5 reload change the behavior) a page embedding a gravatar that's already cached)